tests/helpers/test_event.py

```diff
@@ -334,15 +338,15 @@ def test_track_sunrise(self):
 
         self._send_time_changed(next_rising + offset)
         self.hass.block_till_done()
-        self.assertEqual(2, len(runs))
```

Comment:
OK, I think we're on to something now. After switching to dates, this test started failing. We were previously asserting that the sunrise trigger was run _twice_, which I think is exactly the behavior we were seeing, and want to avoid.